models/DataObject/AbstractObject.php

```diff
@@ -1104,14 +1115,22 @@ public function setUserModification($o_userModification)
     }
 
     /**
-     * @param array $children
+     * @param array|null $children
      *
      * @return $this
      */
     public function setChildren($children)
     {
-        $this->o_children = $children;
-        $this->o_hasChildren = (is_array($children) && count($children) > 0);
+        if($children === null) {
+            // unset all cached children
+            $this->o_children = [];
+            $this->o_hasChildren = [];
+        } elseif(is_array($children)) {
```

Comment:
Is my interpretation correct that setChildren([]) will keep the cached unpublished ones ?

Reply:
you are right.. change to empty check!